### PR TITLE
[DC-599] Abstract dataset access level rendering

### PR DESCRIPTION
**Pull request by s-rubenstein**

### CODE REVIEW

File: src/pages/library/DataBrowserDetails.js [hunk 1]
```diff
@@ -103,24 +104,13 @@ const MainContent = ({ dataObj }) => {
         workspaceName
       ])
     ]),
-    isExternal(dataObj) && div({ style: { marginBottom: '1rem', display: 'flex' } }, [
```
Comment (s-rubenstein) [inline src/pages/library/DataBrowserDetails.js:106]: Removed per conversation with Lou

File: src/pages/library/DataBrowserPreview.js [hunk 2]
```diff
@@ -42,6 +42,38 @@ const styles = {
 
 const activeTab = 'datasets'
 
+const DatasetPreviewSelector = ({ access, dataset, selectedTable, setSelectedTable, selectOptions }) => Utils.switchCase(access,
```
Comment (s-rubenstein) [inline src/pages/library/DataBrowserPreview.js:45]: This abstraction seemed like a nice driveby change. Also cause it wasn't really behaving well as an inline switchCase

File: src/pages/library/dataBrowser-utils.js [hunk 3]
```diff
@@ -86,6 +91,45 @@ export const getDatasetAccessType = dataset => Utils.cond(
...
   () => datasetAccessTypes.CONTROLLED)
 
+export const DatasetAccess = ({ dataset }) => {
+  const { requestingAccess, setRequestingAccess } = useState()
+  const access = getDatasetAccessType(dataset)
+  const { requestAccessURL } = dataset
+  return h(Fragment, [
+    Utils.cond(
+      [!!requestAccessURL && access === datasetAccessTypes.CONTROLLED, () => h(ButtonOutline, {
+        style: { height: 34, textTransform: 'none', padding: '.5rem' },
```
Comment (s-rubenstein) [inline src/pages/library/dataBrowser-utils.js]: Heights changed from unset to a fixed value to help encourage consistency

File: src/pages/library/dataBrowser-utils.js [hunk 4]
```diff
@@ -86,6 +91,45 @@ export const getDatasetAccessType = dataset => Utils.cond(
...
+            title: dataset['dct:title']
+          })
+        }
+      }, [icon('lock'), div({ style: { paddingLeft: 10, fontSize: 12 } }, ['Request Access'])])],
+      [access === datasetAccessTypes.PENDING, () => div({ style: { color: commonStyles.access.pending, display: 'flex', alignItems: 'center' } }, [
+        icon('lock'),
+        div({ style: { paddingLeft: 10, paddingTop: 4, fontSize: 12 } }, ['Pending Access'])
+      ])],
+      [access === datasetAccessTypes.EXTERNAL, () => h(ButtonOutline, {
+        style: { height: 34, textTransform: 'none', padding: '.5rem' },
```
Comment (pshapiro4broad) [inline src/pages/library/dataBrowser-utils.js]: Would it be possible (and worthwhile) to refactor to three cases of `style: { height: 34, textTransform: 'none', padding: '.5rem' }` to avoid the repetition?

**Review by pshapiro4broad — APPROVED** [review]
Looks good, as far as I can tell. It's nice to see some duplicate code being removed.

**Review by lu-c — APPROVED** [review]
This change makes sense! Good to update the common code :)
